runelite-client/src/main/java/net/runelite/client/plugins/statusorbs/StatusOrbsPlugin.java

```diff
@@ -0,0 +1,184 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.statusorbs;
+
+import com.google.inject.Provides;
+import java.awt.image.BufferedImage;
+import java.util.EnumSet;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Affliction;
+import net.runelite.api.Client;
+import net.runelite.api.SpriteID;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.ImageUtil;
+
+@PluginDescriptor(
+	name = "Status Orbs",
+	description = "Settings for the Minimap orbs (HP, Run, & Special Attack)",
+	tags = {"hp", "heart", "minimap", "run", "energy"}
+)
+@Slf4j
+public class StatusOrbsPlugin extends Plugin
+{
+	private static final BufferedImage HEART_DISEASE;
+	private static final BufferedImage HEART_POISON;
+	private static final BufferedImage HEART_VENOM;
+	static
+	{
+		HEART_DISEASE = ImageUtil.resizeCanvas(ImageUtil.getResourceStreamFromClass(StatusOrbsPlugin.class, "1067-DISEASE.png"), 26, 26);
+		HEART_POISON = ImageUtil.resizeCanvas(ImageUtil.getResourceStreamFromClass(StatusOrbsPlugin.class, "1067-POISON.png"), 26, 26);
+		HEART_VENOM = ImageUtil.resizeCanvas(ImageUtil.getResourceStreamFromClass(StatusOrbsPlugin.class, "1067-VENOM.png"), 26, 26);
+	}
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Inject
+	private StatusOrbsConfig config;
+
+	// HeartDisplay
+	private EnumSet<Affliction> currentAfflictions;
+
+	@Provides
+	StatusOrbsConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(StatusOrbsConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		if (config.dynamicHpHeart())
+		{
+			clientThread.invoke(this::checkHealthIcon);
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		if (config.dynamicHpHeart())
+		{
+			clientThread.invoke(this::resetHealthIcon);
+		}
+	}
+
+	@Subscribe
+	private void onConfigChanged(ConfigChanged c)
+	{
+		if (c.getGroup().equals("statusorbs"))
+		{
+			switch (c.getKey())
+			{
+				case "dynamicHpHeart":
+					if (config.dynamicHpHeart())
+					{
+						checkHealthIcon();
+					}
+					else
+					{
+						resetHealthIcon();
+					}
+					break;
+				default:
+					return;
+			}
+		}
+	}
+
+	@Subscribe
+	private void onVarbitChanged(VarbitChanged e)
+	{
+		// Dynamic HP Heart
```

Comment:
This comment doesn't seem very helpful, given `config.dynamicHpHeart()` is pretty self-explanatory.